pageserver/src/walredo.rs

```diff
@@ -626,35 +592,37 @@ impl PostgresRedoProcess {
 
         // Create empty data directory for wal-redo postgres, deleting old one first.
         if datadir.exists() {
-            info!(
-                "old temporary datadir {} exists, removing",
-                datadir.display()
-            );
+            info!("old temporary datadir {datadir:?} exists, removing",);
             fs::remove_dir_all(&datadir)?;
         }
         let pg_bin_dir_path = conf.pg_bin_dir(pg_version).map_err(|e| {
             Error::new(
                 ErrorKind::Other,
-                format!("incorrect pg_bin_dir path: {}", e),
+                format!("incorrect pg_bin_dir path: {e:#}"),
             )
         })?;
         let pg_lib_dir_path = conf.pg_lib_dir(pg_version).map_err(|e| {
             Error::new(
                 ErrorKind::Other,
-                format!("incorrect pg_lib_dir path: {}", e),
+                format!("incorrect pg_lib_dir path: {e:#}"),
             )
         })?;
 
-        info!("running initdb in {}", datadir.display());
-        let initdb = Command::new(pg_bin_dir_path.join("initdb"))
```

Comment:
This one didn't call `current_dir()`, but `prepare_initdb_command` does. Intentional?

Reply:
Yes, I think this way it's safer for the library code to operate: in pageserver's case, we set `workdir` as current dir, and any kinds of creations are more or less ok, but `wal_craft` binary does not do that and might create the directory in the wrong place.